lib/routes/account.js

```diff
@@ -1745,18 +1781,68 @@ module.exports = (
 
         db.account(uid)
           .then(
-            (account) => {
+            (accountResult) => {
+              account = accountResult
+
+              isAccountVerification = butil.buffersAreEqual(code, account.emailCode)
+
               // This endpoint is not authenticated, so we need to look up
               // the target email address before we can check it with customs.
               return customs.check(request, account.email, 'recoveryEmailVerifyCode')
-                .then(
-                  () => account
-                )
             }
           )
+          .then(() => {
+
+            if (! isAccountVerification) {
```

Comment:
I think originally I didn't want to introduce to many changes to this endpoint since it was already pretty hairy with verifying tokens and accounts. However, I think I found a good way to tease apart all of the main logic into something more readable.